packages/commons/src/repositories/EventRepository.ts

```diff
@@ -20,30 +20,26 @@ async function internalCreateEvents<T extends Event>(
   createEvents: Array<CreateEvent<T>>
 ): Promise<string[]> {
   try {
-    return await db.tx(
-      async (t) =>
-        await Promise.all(
-          createEvents.map(async (createEvent) => {
-            const data = await t.oneOrNone(sql.checkEventVersionExists, {
-              stream_id: createEvent.streamId,
-              version: createEvent.version,
-            });
-
-            const newVersion = data != null ? createEvent.version + 1 : 0;
-
-            await t.none(sql.insertEvent, {
-              stream_id: createEvent.streamId,
-              version: newVersion,
-              correlation_id: createEvent.correlationId,
-              type: createEvent.event.type,
-              event_version: createEvent.event.event_version,
-              data: Buffer.from(toBinaryData(createEvent.event)),
-            });
-
-            return createEvent.streamId;
-          })
-        )
-    );
+    await db.tx(async (t) => {
+      for (const createEvent of createEvents) {
+        const data = await t.oneOrNone(sql.checkEventVersionExists, {
+          stream_id: createEvent.streamId,
+          version: createEvent.version,
+        });
+
+        const newVersion = data != null ? createEvent.version + 1 : 0;
+
+        await t.none(sql.insertEvent, {
+          stream_id: createEvent.streamId,
+          version: newVersion,
+          correlation_id: createEvent.correlationId,
+          type: createEvent.event.type,
+          event_version: createEvent.event.event_version,
+          data: Buffer.from(toBinaryData(createEvent.event)),
+        });
+      }
+    });
+    return createEvents.map((createEvent) => createEvent.streamId);
```

Comment:
@MalpenZibo I had to refactor this to create multiple events for the same `stream_id`, otherwise it was not working correctly - with the previous code, it was failing with `Error: Error creating event: error: duplicate key value violates unique constraint "events_stream_id_version_key"`.

Could you review it?

Reply:
Yes! Well using a `Promise.all` was a mistake in the first place so this is more a bugfix than a refactor XD